docs/source/building/variants.rst

```diff
@@ -0,0 +1,911 @@
+
+   requirements:
+       run:
+           - {{ pin_subpackage('my_awesome_subpackage') }}
+
+   outputs:
+     - name: my_awesome_subpackage
+       version: 2.0
+
+by referring to subpackages this way, you don't need to worry about what the end
+version of my_awesome_subpackage will be. Update it independently and just let
+conda build figure it out and keep things consistent. There's more information
+below in the `Referencing subpackages`_ section below.
+
+
+Creating conda-build variant config files
+-----------------------------------------
+
+
+Variant input files are yaml files.
+
+There are some special keys that behave differently and can be more nested:
+
+Search order for these files is the following:
+
+1. a file named ``conda_build_config.yaml`` in the user's HOME folder
+2. an arbitrarily named file specified as the value for the
+   ``conda_build_config`` key in your .condarc file
+3. a file named ``conda_build_config.yaml`` in the same folder as ``meta.yaml``
+   with your recipe
+4. Any additional files specified on the command line with the
+   ``--variant-config-files`` or ``-m`` command line flags, which can be passed
+   multiple times for multiple files. The ``conda build`` and ``conda render``
+   commands accept these arguments.
+
+Files found later in this search order clobber the values from earlier files.
+
+
+Using variants with the conda-build API
+---------------------------------------
+
+
+Ultimately, a variant is just a dictionary. This dictionary is provided directly
+to Jinja2 - you can use any declared key from your variant configuration in your
+Jinja2 templates. There are two ways that you can feed this information into the
+API:
+
+1. pass the ``variants`` keyword argument to API functions. Currently, the
+   ``build``, ``render``, ``get_output_file_path``, and ``check`` functions
+   accept this argument. ``variants`` should be a dictionary with values being
+   lists of versions to iterate over. These are aggregated as detailed in the
+   Aggregation of multiple variants section below.
+
+2. Set the ``variant`` member of a Config object. This is just a dictionary. The
+   values for fields should be strings or lists of strings, except "extended
+   keys", which are documented in the `Extended keys`_ section below.
+
+
+Again, with meta.yaml contents like:
+
+.. code-block:: yaml
+
+   package:
+       name: compiled-code
+       version: 1.0
+
+   requirements:
+       build:
+           - python {{ python }}
+       run:
+           - python {{ python }}
+
+You could supply a variant to build this recipe like so:
+
+.. code-block:: python
+
+   variants = {'python': ['2.7', '3.5']}
+   api.build(path_to_recipe, variants=variants)
+
+
+Special variant keys
+--------------------
+
+
+* ``pin_run_as_build``: should be a dictionary. Keys are package names. Values
+  are "pinning expressions" - explained in more detail in `Customizing
+  compatibility`_. This is a generalization of the ``numpy x.x`` spec.
```

Comment:
Maybe don't mention `numpy x.x` here because later on you point out that it is a bad candidate for using with `pin_run_as_build`.